### DISCUSSION TITLE
Cloudflare R2 (S3 Compatible API) Configuration Example

### DISCUSSION THREAD

**Discussion by nathanaelytj** [laravel/framework#44859] [Q&A] [answered]
How to create config for Cloudflare R2 bucket?

Thank you for help

**Comment by xolf** [comment]
Hey, 

this can be achieved pretty simple. Should I share the configuration with you?

**Reply by nathanaelytj** [reply]
Hello, can you share your sample configuration? It will help me and other looking for example. Thank you

**Comment by xolf** [comment, marked as answer]
```php
// Cloudflare R2 - https://developers.cloudflare.com/r2
// See https://developers.cloudflare.com/r2/examples/aws-sdk-php/ for configuration
'r2' => [
    'driver' => 's3',
    'key' => env('R2_ACCESS_KEY_ID'),
    'secret' => env('R2_SECRET_ACCESS_KEY'),
    'region' => 'auto',
    'bucket' => env('R2_BUCKET'),
    'endpoint' => env('R2_ENDPOINT'),

    'throw' => true,
],
```

**Reply by nathanaelytj** [reply]
Thank you

**Comment by njoguamos** [comment]
Here is an article on [how to setup Cloudflare R2 with Laravel ](https://njoguamos.me.ke/posts/how-to-use-cloudflare-r2-with-laravel-file-storage)
